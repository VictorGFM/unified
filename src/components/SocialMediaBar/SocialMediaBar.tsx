import {
  SocialMediaBarDiv,
  SocialMediaButtonsDiv,
  ToggleSocialMediaBardDiv,
  ToggleSocialMediaBarButton,
} from './styles'
import {
  AppIcon,
  InstagramIcon,
  TwitterIcon,
  FacebookIcon,
  ToggleSocialBarIcon,
} from '../../assets'
import SocialMediaButton from '../SocialMediaButton'
import { useLocation, useNavigate } from 'react-router-dom'

export type SocialType = 'Unified' | 'Instagram' | 'Twitter' | 'Facebook'

const SocialMediaBar = () => {
  const navigate = useNavigate()
  const location = useLocation()

  const currentLocation = location.pathname

  const socialRoutes = {
    Unified: '/app',
    Instagram: '/social/instagram',
    Twitter: '/social/twitter',
    Facebook: '/social/facebook',
  }

  const handleClickButton = (socialType: SocialType) => {
    navigate(socialRoutes[socialType])
  }

  return (
    <SocialMediaBarDiv>
      <SocialMediaButtonsDiv>
        <SocialMediaButton
          icon={AppIcon}
          isSelected={
            currentLocation !== socialRoutes['Instagram'] &&
            currentLocation !== socialRoutes['Twitter'] &&
            currentLocation !== socialRoutes['Facebook']
          }
          onClick={() => handleClickButton('Unified')}
        />
        <SocialMediaButton
          icon={InstagramIcon}
          isSelected={currentLocation === socialRoutes['Instagram']}
          onClick={() => handleClickButton('Instagram')}
        />
        <SocialMediaButton
          icon={TwitterIcon}
          isSelected={currentLocation === socialRoutes['Twitter']}
          onClick={() => handleClickButton('Twitter')}
        />
        <SocialMediaButton
          icon={FacebookIcon}
          isSelected={currentLocation === socialRoutes['Facebook']}
          onClick={() => handleClickButton('Facebook')}
        />
      </SocialMediaButtonsDiv>
      <ToggleSocialMediaBardDiv>
        <ToggleSocialMediaBarButton src={ToggleSocialBarIcon} />
      </ToggleSocialMediaBardDiv>
    </SocialMediaBarDiv>
  )
}

export default SocialMediaBar
