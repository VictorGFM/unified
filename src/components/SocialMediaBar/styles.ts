import styled from 'styled-components'
import { COLORS } from '../../styles/global'

export const SocialMediaBarDiv = styled.div`
  display: flex;
  grid-row: 1/3;
`
export const SocialMediaButtonsDiv = styled.div`
  flex: 1;
`
export const ToggleSocialMediaBardDiv = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;

  width: 17px;

  background-color: ${COLORS.WHITE.ONE};
  box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.48);
  z-index: 1;
`
export const ToggleSocialMediaBarButton = styled.img``
