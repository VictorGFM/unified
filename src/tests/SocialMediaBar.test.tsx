import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
import { describe, expect, it } from 'vitest'
import { render } from '@testing-library/react'
import SocialMediaBar from '../components/SocialMediaBar'

describe('<SocialMediaBar/>', () => {
  it('should show match snapshot', () => {
    const component = render(
      <Router>
        <Routes>
          <Route path="*" element={<SocialMediaBar />} />
        </Routes>
      </Router>
    )

    expect(component).toMatchSnapshot()
  })
})
