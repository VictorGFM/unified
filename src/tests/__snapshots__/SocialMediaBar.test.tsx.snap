// Vitest Snapshot v1

exports[`<SocialMediaBar/> > should show match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bczRLJ eciodL"
      >
        <div
          class="sc-gsnTZi hHFUSO"
        >
          <div
            class="sc-jSMfEi hdIjdc"
            data-cy="social-media-button-unified"
          >
            <img
              class="sc-gKXOVf bWeCPV"
              src="/src/assets/icons/AppIcon.svg"
            />
            <div
              class="sc-iBkjds jvkWpD"
            />
          </div>
          <div
            class="sc-jSMfEi ikLAlr"
            data-cy="social-media-button-instagram"
          >
            <img
              class="sc-gKXOVf bWeCPV"
              src="/src/assets/icons/InstagramIcon.svg"
            />
          </div>
          <div
            class="sc-jSMfEi ikLAlr"
            data-cy="social-media-button-twitter"
          >
            <img
              class="sc-gKXOVf bWeCPV"
              src="/src/assets/icons/TwitterIcon.svg"
            />
          </div>
          <div
            class="sc-jSMfEi ikLAlr"
            data-cy="social-media-button-facebook"
          >
            <img
              class="sc-gKXOVf bWeCPV"
              src="/src/assets/icons/FacebookIcon.svg"
            />
          </div>
        </div>
        <div
          class="sc-dkzDqf juTuKI"
        >
          <img
            class="sc-hKMtZM ixKqbH"
            src="/src/assets/icons/ToggleSocialBarIcon.svg"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bczRLJ eciodL"
    >
      <div
        class="sc-gsnTZi hHFUSO"
      >
        <div
          class="sc-jSMfEi hdIjdc"
          data-cy="social-media-button-unified"
        >
          <img
            class="sc-gKXOVf bWeCPV"
            src="/src/assets/icons/AppIcon.svg"
          />
          <div
            class="sc-iBkjds jvkWpD"
          />
        </div>
        <div
          class="sc-jSMfEi ikLAlr"
          data-cy="social-media-button-instagram"
        >
          <img
            class="sc-gKXOVf bWeCPV"
            src="/src/assets/icons/InstagramIcon.svg"
          />
        </div>
        <div
          class="sc-jSMfEi ikLAlr"
          data-cy="social-media-button-twitter"
        >
          <img
            class="sc-gKXOVf bWeCPV"
            src="/src/assets/icons/TwitterIcon.svg"
          />
        </div>
        <div
          class="sc-jSMfEi ikLAlr"
          data-cy="social-media-button-facebook"
        >
          <img
            class="sc-gKXOVf bWeCPV"
            src="/src/assets/icons/FacebookIcon.svg"
          />
        </div>
      </div>
      <div
        class="sc-dkzDqf juTuKI"
      >
        <img
          class="sc-hKMtZM ixKqbH"
          src="/src/assets/icons/ToggleSocialBarIcon.svg"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
